go/vt/vttablet/onlineddl/executor.go

```diff
@@ -4715,6 +4724,39 @@ func (e *Executor) ForceCutOverPendingMigrations(ctx context.Context) (result *s
 	return result, nil
 }
 
+func (e *Executor) SetMigrationCutOverThreshold(ctx context.Context, uuid string, thresholdString string) (result *sqltypes.Result, err error) {
+	if atomic.LoadInt64(&e.isOpen) == 0 {
+		return nil, vterrors.New(vtrpcpb.Code_FAILED_PRECONDITION, schema.ErrOnlineDDLDisabled.Error())
+	}
+	if !schema.IsOnlineDDLUUID(uuid) {
+		return nil, vterrors.Errorf(vtrpcpb.Code_UNKNOWN, "Not a valid migration ID in FORCE_CUTOVER: %s", uuid)
+	}
+	threshold, err := time.ParseDuration(thresholdString)
+	if err != nil {
+		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "invalid cut-over threshold value: %s. Try '5s' to '30s'", thresholdString)
+	}
+
+	log.Infof("SetMigrationCutOverThreshold: request to set cut-over threshold to %v on migration %s", threshold, uuid)
+	e.migrationMutex.Lock()
+	defer e.migrationMutex.Unlock()
+
+	threshold = safeMigrationCutOverThreshold(threshold)
```

Comment:
IMO it's better to return an error if the user is e.g. trying to set it to 90s to avoid user confusion — especially since this isn't driven by a flag where we can explicitly tell the user about the valid range and also potentially check the input value on the client side. I don't feel strongly about it though (I do see that we're documenting it).

Reply:
We now return an error if the user's value is out of bounds (`< 5s` or `> 30s`). `0` is allowed and presets the threshold value to the factory default.

Reply:
Docs updated.